### DISCUSSION TITLE
Getting Github OAuth Access Token

### DISCUSSION THREAD

**Discussion by emsesc** [nextauthjs/next-auth#1241] [Help] [answered]
I was able to successfully authenticate on the next-auth template with Github, but I need the [access token](https://docs.github.com/en/developers/apps/authorizing-oauth-apps#3-use-the-access-token-to-access-the-api) for developing my app. I believe next-auth only returns name, email, and image in `session`, but can it also handle the Github access token to enable me to make requests to the Github API on behalf of the user?

**Comment by balazsorban44** [comment, marked as answer]
Please refer to the docs, check out the jwt and session callbacks https://next-auth.js.org/configuration/callbacks
